src/modules/collection/utils.ts

```diff
@@ -49,3 +49,27 @@ export function toCollectionObject(collections: Collection[]) {
     return obj
   }, {} as Record<string, Collection>)
 }
+
+export function canSeeCollection(collection: Collection, address: string) {
```

Comment:
There's a lib file named address that has a few utils for these address comparisons. You can use isEqual from there!